### DISCUSSION TITLE
nim.cfg but for Nelua?

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#67] [Q&A] [answered]
With Nim we can use custom settings with the help of `nim.cfg` in a project's directory.

I would like to use `ccache gcc` at a global usage; I know it works with `nelua --cc="ccache gcc"` which helps reducing the compilation time exponentially, so I would like it to be my default setting.

Does Nelua support custom config settings somehow?

**Comment by edubart** [comment, marked as answer]
Yes, sorry as this is not in the documentation yet, but this is supported, you can create `neluacfg.lua` files in the following directories:

* `$HOME/.config/nelua/neluacfg.lua` for global configurations across the system
* `.neluacfg.lua` in the current directory `nelua` is running from for per project configuration

The compiler will give priority for command line flags, then `.neluacfg.lua` in the current directory and finally fallback to `$HOME/.config/nelua/neluacfg.lua` for global configuration in the system.

The following is an example of its contents:

```lua
return {
  cc="ccache gcc",
  cflags="-Wall -Wextra",
  add_path={
    '/home/bart/projects/nelua/nelua-decl/libs/?/?.nelua',
  }
}
```

The file can have most options from `nelua --help`, but you need to use the full option name and replace any `-` with `_`.

**Reply by stefanos82** [reply]
Very nice! I will try it right now. Cheers :+1:

**Reply by stefanos82** [reply]
An interesting behavior takes place. With 

```
return {
  cc="ccache gcc",
  cflags="-Wall -Wextra",
}
```
it generates the following code: `/* Compile command: ccache gcc "nelua_cache/auto.c" -o "nelua_cache/auto" -Wall -fwrapv -g -Wall -Wextra -lm */`

**Reply by edubart** [reply]
This is expected, the compile command is always saved as a comment in the source file, thus the generated C file will have a different hash when changing the cflags or C compiler, thus Nelua won't use any cached compilation whenchanging the cflags or C compile. Nelua compiler need to save the compile command somewhere for this, and it's saved in the source, also saving there hints any user reading the C source what compiler and flags where used to compile the source in question.

**Reply by stefanos82** [reply]
That's not what I meant. `-Wall` is found twice and the weird thing to me is that it places my own flags right before the end of `-Wall -fwrapv -g -lm`, between `-g` and `-lm` to be more precise.

**Reply by edubart** [reply]
Oh, it's because `-Wall` is enabled by default  at the moment and adding it in the `neluacfg.lua` was not really needed, just added as an example, so you can remove it.

**Reply by stefanos82** [reply]
Ah OK. I've just tested it; works as expected :partying_face:
